### PR TITLE
qt: fix hardcoded versioned GCC references on Linux

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Also, while we're here, let's fix up a faulty shim path `inreplace`,
since it leaves a stray `/` in front of `clang` and `clang++`.

Fixes a test failure in #106755.

### CODE REVIEW

**Comment by bayandin** [comment]
Need to rebase

File: Formula/qt.rb [hunk 1]
```diff
@@ -129,31 +129,30 @@ class Qt < Formula
...
-    resources.each do |r|
-      venv.pip_install r
-    end
-    xy = Language::Python.major_minor_version "python3"
-    ENV.prepend_path "PYTHONPATH", buildpath/"venv/lib/python#{xy}/site-packages"
-    ENV.prepend_path "PATH", Formula["python@3.10"].libexec/"bin"
+    venv.pip_install resources
+
+    ENV.prepend_path "PYTHONPATH", buildpath/"venv"/Language::Python.site_packages("python3")
+    ENV.prepend_path "PATH", Formula["python@3.10"].opt_libexec/"bin"
```
Comment (cho-m) [inline Formula/qt.rb:135]: Maybe set `PATH` before other `"python3`" usage to help in next Python 3.11 migration?

Comment (cho-m) [reply Formula/qt.rb:135]: I guess doesn't really matter if https://github.com/Homebrew/brew/pull/13677 is merged

**Comment by iMichka** [comment]
Rebased

**Comment by yurikoles** [comment]
`qt@5` is also affected.

File: Formula/qt.rb [hunk 2]
```diff
@@ -254,6 +255,18 @@ def install
     end
   end
 
+  def post_install
+    return if OS.mac?
+
+    # Ensure the hard-coded versioned `gcc` reference does not go stale.
+    ohai "Fixing up GCC references..."
+    gcc_version = Formula["gcc"].any_installed_version.major
+    inreplace(share/"qtbase/mkspecs/common/g++-base.conf") do |s|
```
Comment (carlocab) [inline Formula/qt.rb:264]: This path is still wrong.

File: Formula/qt.rb [hunk 3]
```diff
@@ -128,32 +128,31 @@ class Qt < Formula
 
   def install
     # Install python dependencies for QtWebEngine
-    venv = virtualenv_create(buildpath/"venv", "python3")
-    resources.each do |r|
-      venv.pip_install r
-    end
-    xy = Language::Python.major_minor_version "python3"
-    ENV.prepend_path "PYTHONPATH", buildpath/"venv/lib/python#{xy}/site-packages"
```
Comment (carlocab) [inline Formula/qt.rb:136]: QtWebEngine didn't build, so maybe this is still needed for some reason? No idea why though, given that the venv Python is first in `PATH`.

Also, it's weird that it just silently did not build.

File: Formula/qt.rb (hunk 2 above)
Comment (cho-m) [reply Formula/qt.rb:264]: Maybe:
```suggestion
    inreplace(pkgshare/"mkspecs/common/g++-base.conf") do |s|
```

We do pass in `-DINSTALL_MKSPECSDIR=share/qt/mkspecs`

Comment (carlocab) [reply Formula/qt.rb:264]: I was pretty sure I checked the bottle for the right path though. Must've read it wrong.

File: Formula/qt.rb (hunk 3 above)
Comment (cho-m) [reply Formula/qt.rb:136]: CMake build scripts are picking up `HOMEBREW_PREFIX/bin/python3`. In `CMakeCache.txt`:
```
//Path to a program.
Python3_EXECUTABLE:FILEPATH=/usr/local/bin/python3
```

Don't know why.

Not sure if there is a way to fail in this scenario. From https://github.com/qt/qtwebengine/blob/v6.3.1/src/CMakeLists.txt#L20-L36, it looks like related cache variables are going to be force modified based on upstream CI needing to pass configure step.

Comment (carlocab) [reply Formula/qt.rb:136]: I wonder if CMake is detecting the first `python[3]` it finds as belonging in a venv so skips it. That would explain why we need `PYTHONPATH`.

**Comment by carlocab** [comment]
Removing the `long build` label so we get a run-attempt on ARM.

**Comment by carlocab** [comment]
Ah whoops. Restarted Linux but I forgot that that also restarts all the Intel runners.

**Pull request closed without merging**